### PR TITLE
Email person from ABB

### LINKED ISSUE #10: Emailing someone from AB3 is difficult
After finding the contact information, I am unable to email the person.

As a user, I should be able to email the person of my choice.

### PR DESCRIPTION
**Pull request by yaowei-soc**
Closes #10, #135, #136

Currently implemented:

- [x] Email person(s) in person list via OS email client

Will trigger a `mailto` url containing the emails of the selected persons.

Reason for decreasing code coverage:
- private `getEmailString()` in `EmailCommand` and `getHostServices().showDocument()` untestable/hard to test.

@tanboonji Documentation
@oeiyiping Documentation/Model/Commands
@darkdestry-t Overall implementation

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -234,6 +234,23 @@ Examples:
 |`list -a `| show the contact's name and address only.|
 |`list -a -p`| show the contact's name, address and phone number only.|
 
+### Email Command
+
+Email command provides the user with the choice to mass email person(s) of their choice.
+Executing the `email` command will open up the operating system's email client.
+
+Format: `email [INDEX...] [shown]`
```
Comment (oeiyiping) [inline docs/UserGuide.md:242]: Is `[INDEX...]` meant to signify multiple index arguments? If so, maybe change it to `[INDEX]...` instead, to comply with the format defined in line 53.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -234,6 +234,23 @@ Examples:
...
+
+`shown` is a special index that refers to all the person that are in the visible person list.
```
Comment (oeiyiping) [inline docs/UserGuide.md:244]: Would it be better to include this line as part of "Notes on `email` command:", to be consistent with the rest of the commands?

File: src/main/java/seedu/address/logic/commands/EmailCommand.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ * Opens email client with the selected person(s) email as the subject.
+ */
+public class EmailCommand extends Command {
+
+    public static final String COMMAND_WORD = "email";
+    public static final String MESSAGE_SUCCESS = "Opened email client";
+    public static final String MESSAGE_FAILURE = "URL Exception occurred";
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Open email client, with email subjects.\n"
+                    + "Parameters: [INDEX ...]\n"
```
Comment (oeiyiping) [inline src/main/java/seedu/address/logic/commands/EmailCommand.java:28]: Same formatting matter as mentioned in the other comment ([INDEX]...).

**Review by oeiyiping — CHANGES_REQUESTED** [review]
Looks good overall, aside from the following matters:

1. Out-of-bounds index does not return any error message in ABB, and there is no other indication that the input index is invalid.
2. Some minor docs issues.
3. Just a thought: Would it be good to include a disclaimer in ABB itself and/or in the UG stating that the email command is only valid if the user has an OS mail app, and is logged into a valid account in said mail app?

**Comment by yaowei-soc** [comment]
> 
> 
> Looks good overall, aside from the following matters:
> 
>     1. Out-of-bounds index does not return any error message in ABB, and there is no other indication that the input index is invalid.
>        Might be good to reference the error behaviour in `delete` command for consistency.
> 
>     2. Some minor docs issues.
> 
>     3. I feel that it would it be good to include a disclaimer in ABB itself and/or in the UG stating that the `email` command is only valid if the user has an OS mail app, and is logged into a valid account in said mail app. What do you think?

For point 3, I'm not exactly sure that is within our control? I've tried on Windows (logged in), macOS (logged in), Ubuntu (not logged in, Thunderbird installed) and on Ubuntu.

On Ubuntu, it will first open up the default browser (FireFox) and subsequently opening up Thunderbird, prompting me to login.

image

So I believe that on most OS, if the user is not logged in, the mail client will prompt the user. I feel that I don't have to document that down.

File: docs/UserGuide.md (hunk 2 above)
Comment (yaowei-soc) [reply docs/UserGuide.md]: Updated in the latest commits

**Review by oeiyiping — APPROVED** [review]
Aside from the minor point on disclaimer, LGTM.

**Review by darkdestry-t — APPROVED** [review]
Looks like you're just letting OS handle the URI for the email function. I dont see any other way to implement it natively. Looks good. LGTM

**Review by tanboonji — APPROVED** [review]

**Review by darkdestry-t — COMMENTED** [review]
Looks ok except the missing test case for different partial collection during equality check

**Pull request merged**